railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
Ok, I see this is because of https://github.com/vipulnsward/rails/commit/674dab30bc46d63875c533f4cd4daa2b62ad67c5 by @jmbejar 

I think we should not do that by default. Else we are going to see surprising behaviour like - https://github.com/rails/rails/issues/24264